pkg/virt-launcher/virtwrap/network/dhcpv6/dhcpv6.go

```diff
@@ -108,23 +117,19 @@ func (h *DHCPv6Handler) ServeDHCPv6(conn net.PacketConn, peer net.Addr, m dhcpv6
 			response, err = dhcpv6.NewReplyFromMessage(dhcpv6Msg, h.modifiers...)
 		}
 	default:
-		log.Log.V(4).Info("DHCPv6 - non Solicit request recieved")
+		log.Log.V(4).Info("DHCPv6 - non Solicit request received")
 		response, err = dhcpv6.NewReplyFromMessage(dhcpv6Msg, h.modifiers...)
 	}
 
-	ianaRequest := msg.Options.OneIANA()
-	ianaResponse := response.Options.OneIANA()
-	ianaResponse.IaId = ianaRequest.IaId
-	response.UpdateOption(ianaResponse)
-
 	if err != nil {
-		log.Log.V(4).Errorf("DHCPv6 failed sending a response to the client: %v", err)
-		return
+		return nil, err
 	}
 
-	if _, err := conn.WriteTo(response.ToBytes(), peer); err != nil {
-		log.Log.V(4).Errorf("DHCPv6 cannot reply to client: %v", err)
-	}
+	ianaRequest := dhcpv6Msg.Options.OneIANA()
+	ianaResponse := response.Options.OneIANA()
+	ianaResponse.IaId = ianaRequest.IaId
+	response.UpdateOption(ianaResponse)
+	return response, nil
```

Comment:
`buildResponse` method was introduced so that it will be possible to verify the built response with using a mock. I think that even without this reason, it is good to have `buildResponse` as a separate method.
But this is the reason why this code is here.

Reply:
Fair enough.

Reply:
The only production code change is moving the response building to a method. This was done especially for the unit tests to avoid usage of mock.
But since I don't want to spend time on arguing about this (IMO, non important issue). I will move the production code to another commit.